fidesctl/src/fidesctl/core/system.py

```diff
@@ -64,13 +64,95 @@ def generate_redshift_systems() -> List[System]:
     return redshift_systems
 
 
+def describe_rds_clusters() -> Dict[str, List[Dict]]:
+    """
+    Creates boto3 rds client and returns describe_db_clusters response.
+    """
+    rds_client = boto3.client(
+        "rds",
+    )
+    describe_clusters = rds_client.describe_db_clusters()
+    return describe_clusters
+
+
+def describe_rds_instances() -> Dict[str, List[Dict]]:
+    """
+    Creates boto3 rds client and returns describe_db_instances response.
+    """
+    rds_client = boto3.client(
+        "rds",
+    )
+    describe_instances = rds_client.describe_db_instances()
+    return describe_instances
+
+
+def transform_rds_systems(
+    describe_clusters: Dict[str, List[Dict]], describe_instances: Dict[str, List[Dict]]
+) -> List[System]:
+    """
+    Given a describe clusters and describe instances responses, build a system object
+    which represents each cluster or instance.
+
+    A system is created for each cluster, but for instances we only create a system if
+    it is not part of a cluster
+    """
+    rds_cluster_systems = [
+        System(
+            fides_key=cluster["DBClusterIdentifier"],
+            name=cluster["DBClusterIdentifier"],
+            description=f"Fides Generated Description for RDS Cluster: {cluster['DBClusterIdentifier']}",
+            system_type="rds_cluster",
+            fidesctl_meta=SystemMetadata(
+                endpoint_address=cluster["Endpoint"],
+                endpoint_port=cluster["Port"],
+                resource_id=cluster["DBClusterArn"],
+            ),
+            privacy_declarations=[],
+        )
+        for cluster in describe_clusters["DBClusters"]
+    ]
+    rds_instances_systems = [
+        System(
+            fides_key=instance["DBInstanceIdentifier"],
+            name=instance["DBInstanceIdentifier"],
+            description=f"Fides Generated Description for RDS Instance: {instance['DBInstanceIdentifier']}",
+            system_type="rds_instance",
+            fidesctl_meta=SystemMetadata(
+                endpoint_address=instance["Endpoint"]["Address"]
+                if instance.get("Endpoint")
+                else None,
+                endpoint_port=instance["Endpoint"]["Port"]
+                if instance.get("Endpoint")
+                else None,
+                resource_id=instance["DBInstanceArn"],
+            ),
+            privacy_declarations=[],
+        )
+        for instance in describe_instances["DBInstances"]
+        if not instance.get("DBClusterIdentifier")
+    ]
+    return rds_cluster_systems + rds_instances_systems
+
+
+def generate_rds_systems() -> List[System]:
+    """
+    Fetches RDS clusters and instances from AWS and returns the transformed Sytem representations.
+    """
```

Comment:
My only nit is that I'd like to see the client created here and then passed into the functions so it was more transparent, but seeing how boto3 works this current solution seems totally fine to me, just food for thought

Reply:
sure, if we see the need for it I'm open to changing this